### DISCUSSION TITLE
Please help me understand the information given to me

### DISCUSSION THREAD

**Discussion by NewJrIT** [PartialVolume/shredos.x86_64#261] [Q&A] [answered]
I am using shredos on old hard drives before I dump them, and I have noticed a trend with a few of them and I'm not sure what it means. First of all, there is the 35C-40C range of numbers while the disks are being shredded. I assume it is a temperature thing, but for one hard drive the number was flashing red at around 40C with the other two hard drives bouncing around that number as well. I also had one fail and have a value of -1752481C and it was flashing a deeper red.

**Comment by PartialVolume** [comment, marked as answer]
>First of all, there is the 35C-40C range of numbers while the disks are being shredded. I assume it is a temperature thing

Yes, that's correct, that's the actual drive temperature and is sampled every 30 seconds, if I remember correctly. The vast majority of hard drives have a max operating temperature of about 60-70 degrees C. In addition to the current temperature of the drive, ShredOS (nwipe) will try to obtain the temperature limits from the drive but not all drives provide this information. The flashing red is just an indication that a supposed limit has been reached. No action is taken by nwipe, i.e it doesn't attempt to slow the drive down to reduce temperature, it's simply a warning. In the case of the flashing red at 40 deg.C that is obviously incorrect. Either the drive produced incorrect data or you are running an older version of ShredOS, i.e not nwipe v0.37 as I seem to remember there was a bug related to temperature limits. So make sure you are running the latest ShredOS.

>I also had one fail and have a value of -1752481C and it was flashing a deeper red.

If the drive failed, sounds like it was providing corrupt data for the temperature, assuming you are running the latest ShredOS.

If you look at the nwipe log file, you will find a couple of lines that show the max, min, critical and low critical temperatures of the drive. This limit data is provided by the kernels hwmon library.
```
[2024/07/08 23:12:48]  notice: hwmon: sda has temperature monitoring
[2024/07/08 23:12:48]    info: Temperature limits for /dev/sda, critical=N/A, max=N/A, highest=N/A, lowest=N/A, min=N/A, low critical=N/A.
```
In the above example no temperature limits were obtained from the drive via hwmon so the temperature in nwipe will remain as white text. Red flashing means max operating temperature reached, non flashing red means critical high temperature reached. Black flashing text means minimum operating temperature reached, black non flashing means critical low temperature reached.

It's o.k to operate the drive close to its max operating temperature, although obviously better for the drives longevity  if it's cooler. Critical temperature means damage to the drive could occur.

In a well ventilated desktop machine a drive doing nothing will have a temperature of about 25-35 deg.C, however while nwipe is wiping the drive, the temperature may rise to 40-50 deg.C. The ambient temperature of the room makes a difference too, So a room temperature of 30 deg.C will raise the drive temperature even higher compared to a air-conditioned room at 20 degC. Drives in laptops can often get hotter due to the confined space or heatsinks blocked full of dust.
